core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -239,12 +209,12 @@ private[spark] object ClosureCleaner extends Logging {
       cleanTransitively: Boolean,
       accessedFields: Map[Class[_], Set[String]]): Unit = {
 
-    // most likely to be the case with 2.12, 2.13
+    // indylambda check. Most likely to be the case with 2.12, 2.13
     // so we check first
     // non LMF-closures should be less frequent from now on
-    val lambdaFunc = getSerializedLambda(func)
+    val maybeIndylambdaProxy = IndylambdaScalaClosures.getSerializationProxy(func)
```

Comment:
Hi, @rednaxelafx . Do we need to change this PR for Scala 2.11 when we backport into `branch-2.4`?

Reply:
Never mind. I found your previous comment (https://github.com/apache/spark/pull/28463#issuecomment-624779079).

Reply:
Right, the new `IndylambdaScalaClosures.getSerializationProxy` is pretty much the same as the old `getSerializedLambda`, just with a few more checks.
It's supposed to return `None` for Scala 2.11 closures, and `Some(...)` for Scala 2.12+ closures, no changes.